### DISCUSSION TITLE
Include link to rule documentation in LSP diagnostics

### DISCUSSION THREAD

**Discussion by rchl** [rome/tools#4093] [Suggestions]
It would be useful to have a link to the documentation of the relevant rule for diagnostics reported when running in LSP mode. For example here we see a diagnostic specifying a rule name but we can't click the rule name to show more information about the rule.

![Screenshot 2022-12-22 at 21 15 08](https://user-images.githubusercontent.com/153197/209218616-19ee666c-4553-4347-a2d8-2f06b3f46edc.png)

Here is an example of a different diagnostics from eslint with a documentation link:

![Screenshot 2022-12-22 at 21 16 40](https://user-images.githubusercontent.com/153197/209218767-1bc82c86-ead4-454d-9391-d85d7cbdea70.png)

Per [LSP documentation](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/), the `Diagnostic` object can specify `codeDescription` property which includes an URL. This can be used for this purpose and this is what eslint uses. For example:

```json
{
  "diagnostics": [
    {
      "code": "space-in-parens",
      "codeDescription": {
        "href": "https://eslint.org/docs/rules/space-in-parens"
      },
      "message": "There should be no space before this paren.",
      "range": {
        "end": {
          "character": 41,
          "line": 1019
        },
        "start": {
          "character": 40,
          "line": 1019
        }
      },
      "severity": 1,
      "source": "eslint"
    }
  ],
  "uri": "...ts"
}
```

**Comment by rchl** [comment]
Note that eslint also provides an extra code action for opening the rule documentation link but this does not necessarily belong in the same issue. Also, it's a bit special because it needs custom editor command to trigger opening of the link in the browser.

![Screenshot 2022-12-22 at 21 29 49](https://user-images.githubusercontent.com/153197/209220720-54618175-3546-41a6-985e-46ef8fd621f1.png)

**Comment by ematipico** [comment]
@rchl for feature requests, we use GitHub discussions :)

**Reply by rchl** [reply]
Would it make sense for the [new issue template](https://github.com/rome/tools/issues/new/choose) to have `Feature request` entry that links to discussions? :)
